components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,337 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCFontScaler.h"
+
+#import <objc/runtime.h>
+
+#import "UIApplication+AppExtensions.h"
+#import "UIFont+MaterialScalable.h"
+#import "private/MDCFontTraits.h"
+
+MDCTextStyle const MDCTextStyleHeadline1 = @"MDC.TextStyle.Headline1";
```

Comment:
What's the advantage of using `NSString *` instead of NSIntegers here?  In the code below, it becomes a series of `if/else` statements, which are often far slower than `switch`.

Reply:
String are much more extensible.  If they are integers and Geo adds some styles and YouTube adds some styles, they chance of a collision is high.